server.js

```diff
@@ -27,7 +19,54 @@ app.get('/', (req, res) => {
   res.send('Hello world')
 })
 
+const PER_PAGE = 10
+
+app.get('/books', (req, res) => {
+  const { page } = req.query
+  const startIndex = PER_PAGE * +page
+  let booksData = books
+
+  const { lang } = req.query
+  const { title } = req.query
+  const { rating } = req.query
+
+
+  if (rating) {
+    if (rating === 'asc') {
+      booksData.sort(function (a, b) { return a.average_rating - b.average_rating })
+    } else if (rating === 'desc') {
+      booksData.sort(function (a, b) { return b.average_rating - a.average_rating })
+    }
+  } else {
+    booksData.sort(function (a, b) { return a.bookID - b.bookID })
+  }
+
+  if (lang) {
+    booksData = booksData.filter(book => book.language_code === lang)
+  }
+
+  if (title) {
+    booksData = booksData.filter(book => book.title.toString().toLowerCase().includes(title.toLowerCase()))
+  }
+  const length = booksData.length
+  booksData = booksData.slice(startIndex, startIndex + PER_PAGE)
+
+  res.json({
+    totalPages: Math.floor(length / PER_PAGE),
+    currentPage: +page,
+    booksData
+  })
+})
+
+app.get('/books/:id', (req, res) => {
+  const id = req.params.id
+
+  const book = books.filter((book) => book.bookID === +id)
+
+  res.send(book)
```

Comment:
I'm not actually sure in this instance if there's any difference since express is smart enough to see you're sending a JS object here and so it makes it a json response, but you should aim to be explicit and use `res.json(...)`